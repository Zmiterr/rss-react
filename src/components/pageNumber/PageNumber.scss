.pageNumber {
	margin-top: 15px;
}
