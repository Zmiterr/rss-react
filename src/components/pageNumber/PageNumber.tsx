import React, { ReactElement } from 'react';
import { useDispatch } from 'react-redux';
import { IPageNumberProps } from '../../types/types';
import { setPageAct } from '../../types/actions';

import './PageNumber.scss';

const PageNumber = ({ page }: IPageNumberProps): ReactElement => {
	const dispatch = useDispatch();

	const setPage = (num: number) => {
		dispatch(setPageAct(num));
	};
	return (
		<div className="pageNumber">
			<label className="" htmlFor="pageNumber">
				Page:
				<input
					className="form__input"
					type="number"
					name="pageNumber"
					onChange={(e) => setPage(Number(e.target.value))}
					value={page}
					min="1"
				/>
			</label>
		</div>
	);
};

export default PageNumber;
