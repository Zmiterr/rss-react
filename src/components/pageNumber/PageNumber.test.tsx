import React from "react";
import PageNumber from './PageNumber';
//test deps
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from 'react-router-dom';



Enzyme.configure({ adapter: new Adapter() })

describe("PageNumber component", () => {
	it("should render PageNumber component", () => {
		const component = shallow(<BrowserRouter><PageNumber  page={1}/></BrowserRouter>);
		expect(component).toMatchSnapshot();
	});
});

