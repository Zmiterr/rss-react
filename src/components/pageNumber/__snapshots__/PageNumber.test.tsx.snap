// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNumber component should render PageNumber component 1`] = `ShallowWrapper {}`;
